File: LeetCode75/Level1/Find_Pivot_Index/Find_Pivot_Index.py
# 724. Find Pivot Index Python
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        left = 0
        right = 0
        for num in nums:
            # if index == 0:
            #     continue    
            right += num
            
        for index, num in enumerate(nums):
            right -= num
            if left == right:
                return index
            else:
                left += num

        return -1
        

File: LeetCode75_LevelOne/Find_Pivot_Index/Find_Pivot_Index.cc
// 724. Find Pivot Index C++
#include <vector>

class Solution {
public:
    int pivotIndex(std::vector<int>& nums) {
        int left = 0;
        int right = 0;
        for (auto num : nums) right+=num;
        int len = nums.size();
        for (int i = 0; i < len; i++) {
            int num = nums.at(i);
            right -= num;
            if (left == right) return i;
            left += num;
        }
        return -1;
    }
};